### DISCUSSION TITLE
오버로딩과 오버라이딩의 차이에 대해 설명해주세요.

### DISCUSSION THREAD

**Discussion by J-I-H-O** [2024-woowacourse-study/level-interview#85] [Java]
.

**Comment by J-I-H-O** [comment]
오버로딩은 같은 메서드 이름을 가지지만, 메서드 시그니처(파라미터의 타입, 개수, 순서 등)이 다른 여러 메서드를 정의하는 것을 말합니다.

오버라이딩은 상위 클래스의 메서드를 하위 클래스에서 다시 정의하는 것입니다. 상속 관계에 있을 때만 발생합니다.

**Comment by 3Juhwan** [comment]
오버로딩은 같은 이름의 메소드를 여러 개 정의하는 것이며, 메소드 시그니처(매개변수의 개수, 순서, 타입)가 서로 달라야 합니다. 이는 메소드 호출 시에 전달되는 인자에 따라 다르게 동작하도록 합니다.

오버라이딩은 상속 관계에서 부모 클래스의 메소드를 하위 클래스에서 재정의하는 것입니다. 메소드의 이름, 매개변수, 반환 타입이 부모 클래스의 메소드와 일치해야 하며, 이를 통해 다형성을 구현할 수 있습니다. 따라서 동일한 메소드 호출에 대해 객체의 실제 타입에 따라 적절한 메소드가 실행됩니다.
